### PR TITLE
Prefix tree based alias matching

### PR DESCRIPTION
**Pull request by raouldeheer**
## tl;dr
Added support for multiple fall-back locations.
Added prefix-tree based alias matching, this improves performance between 20-80%.

## Issues
This pull request doesn't close any issue. It's just a performance improvement. And adding support for multiple fall-back locations.

## Changes
- Added support for multiple fall-back locations.
- Added prefix-tree based alias matching.

## Conclusion
In the performance test is shown that using a prefix tree to find and resolve aliases is much faster.

### Performance tests:
All results are an average of 5 test runs.
Test were performed with [tsc-alias-benchmark](https://github.com/raouldeheer/tsc-alias-benchmark).
Test specs:
- Ryzen 2700X CPU
- 32GB Ram
- 500GB Samsung 970 Evo SSD (With only the benchmark)
- Node 17.1.0

### Results:
[prefix-tree-matching_performance-data.pdf](https://github.com/justkey007/tsc-alias/files/7610768/prefix-tree-matching_performance-data.pdf)

### CODE REVIEW

**Comment by justkey007** [comment]
Hi @raouldeheer ,
thank you very much for your contribution. You do a very good job
